helix-core.el

```diff
@@ -229,6 +247,17 @@ previous character before moving to the previous long word."
     (set-mark-command nil)
     (end-of-line)))
 
+(defun helix-select-line-up ()
+  "Select the current line, extending upward on every subsequent call."
+  (interactive)
+  (if (and (region-active-p) (bolp))
+      (progn
+        (call-interactively #'previous-line)
+        (beginning-of-line))
+    (end-of-line)
+    (set-mark-command nil)
```

Comment:
nit: Helix Mode inconsistently uses `set-mark-command` and `push-mark`, but I think we should be preferring `push-mark` from here on out so we can suppress the "Mark activated" message during navigation.

e.g. `(push-mark-command t t)`